### ISSUE TITLE
(check-unsat-cores) Segmentation fault

### ISSUE THREAD

**Issue by muchang** [cvc5/cvc5#5902]
```
[551] % cvc4 --check-unsat-cores -i -q small.smt2 
unsat
CVC4 suffered a segfault.
Offending address is 0x30
Looks like a NULL pointer was dereferenced.
Aborted
[552] % 
[552] % cat small.smt2 
(check-sat false)
(get-unsat-core)
[553] %
```
Commit: d47a870

**Comment by nafur** [comment]
In debug, this trigger the following assertion:
```
Fatal failure within const ResolutionChain& CVC4::TSatProof<Solver>::getResolutionChain(CVC4::ClauseId) const [with Solver = CVC4::Minisat::Solver; CVC4::TSatProof<Solver>::ResolutionChain = CVC4::ResChain<CVC4::Minisat::Solver>; CVC4::ClauseId = unsigned int] at /home/gereon/cvc4/src/./proof/sat_proof_implementation.h:435
Check failure

 hasResolutionChain(id)
```

**Comment by HanielB** [comment]
Closed via #6574
